test/e2e/scheduler_predicates.go

```diff
@@ -110,6 +110,20 @@ func verifyResult(c *client.Client, podName string, ns string) {
 			"reason":                   "FailedScheduling",
 		}.AsSelector())
 	expectNoError(err)
+	// If we failed to fined event with a capitalized first letter of reason
+	// try looking for one starting with a small one for backward compatibility.
```

Comment:
Can you please add an explanation for why this block exists, (with a citation of #15806)?  And can you also please file an issue for removing support for this, and add a TODO here to remove this check?